### PR TITLE
1.3 transport cleanups

### PR DESCRIPTION
**Pull request by kseager**
1) took out long_frag usage from OFI transport
2) refactored nbi pathway in transport
3) added nbi to portals4 transport
resolves issue #91

### CODE REVIEW

**Comment by jdinan** [comment]
Excellent work!  I'll run this through testing on the Portals IB build.  Apart from that final check, I think this is ready to merge.

File: src/transport_portals4.h [hunk 1]
```diff
@@ -534,6 +534,80 @@ shmem_transport_put_nb(void *target, const void *source, size_t len,
...
+                     pt,
+                     0,
+                     offset,
+                     NULL,
+                     0);
+        if (PTL_OK != ret) { RAISE_ERROR(ret); }
+
+    } else {
+        shmem_transport_portals4_long_frag_t *long_frag;
+        ptl_handle_md_t md;
```
Comment (jdinan) [inline src/transport_portals4.h]: ```md``` and ```long_frag``` are unused.  Could you add a patch to delete them?

**Comment by kseager** [comment]
thanks! =). 
added in the md/long_frag var clean-up. Thanks for the catch. I wanted roll it in a previous commit...but left it as a stand alone commit.

**Comment by jdinan** [comment]
Thanks!  SOS unit tests and UH test suite passed using SOS Portals 4 build on the IB cluster.

Merging..

**Pull request merged**